easybuild/tools/build_log.py

```diff
@@ -103,25 +112,44 @@ def deprecated(self, msg, max_ver):
 
     def nosupport(self, msg, ver):
         """Print error message for no longer supported behaviour, and raise an EasyBuildError."""
-        self.error("NO LONGER SUPPORTED since v%s: %s; see %s for more information" % (ver, msg, DEPRECATED_DOC_URL))
+        nosupport_msg = "NO LONGER SUPPORTED since v%s: %s; see %s for more information"
+        raise EasyBuildError(nosupport_msg, ver, msg, DEPRECATED_DOC_URL)
 
     def error(self, msg, *args, **kwargs):
         """Print error message and raise an EasyBuildError."""
-        newMsg = "EasyBuild crashed with an error %s: %s" % (self.caller_info(), msg)
-        fancylogger.FancyLogger.error(self, newMsg, *args, **kwargs)
+        ebmsg = "EasyBuild crashed with an error %s: " + msg
+        args = (self.caller_info(),) + args
+
+        fancylogger.FancyLogger.error(self, ebmsg, *args, **kwargs)
+
         if self.raiseError:
-            raise EasyBuildError(newMsg)
+            self.deprecated("Use 'raise EasyBuildError' rather than error() logging method that raises", '3.0')
+            raise EasyBuildError(ebmsg, *args)
+
+    def _error_no_raise(self, msg):
+        """Utility function to log an error with raising an exception."""
+        # make sure raising of error is disabled
+        orig_raise_error = self.raiseError
+        self.raiseError = False
+
+        self.deprecated("Use of dedicated _error_no_raise log method", '3.0')
+        self.error(msg)
+
+        # reinstate previous raiseError setting
+        self.raiseError = orig_raise_error
 
     def exception(self, msg, *args):
         """Print exception message and raise EasyBuildError."""
         # don't raise the exception from within error
-        newMsg = "EasyBuild encountered an exception %s: %s" % (self.caller_info(), msg)
+        ebmsg = "EasyBuild encountered an exception %s: " + msg
+        args = (self.caller_info(),) + args
```

Comment:
no need, `args` is a tuple already, see https://docs.python.org/2/tutorial/controlflow.html#arbitrary-argument-lists